diff --git a/dist/bindable.mjs b/dist/bindable.mjs
index 9df089eb078c3c96c9ad0ee6a013aaa4104dcf67..b6409627e2fe0082470359c86abd55a7d8f2bc70 100644
--- a/dist/bindable.mjs
+++ b/dist/bindable.mjs
@@ -18,7 +18,7 @@ function bindable(props) {
       return controlled() ? props().value : store.value;
     },
     set(nextValue) {
-      const prev = store.value;
+      const prev = this.get();
       const next = isFunction(nextValue) ? nextValue(prev) : nextValue;
       if (props().debug) {
         console.log(`[bindable > ${props().debug}] setValue`, { next, prev });
